java/Anton-Kozyk/src/com/zoolatech/lecture3/tasks/_1/Musician.java

```diff
@@ -0,0 +1,85 @@
+package com.zoolatech.lecture3.tasks._1;
+
+import java.util.Objects;
+
+abstract public class Musician {
+    String name;
+
+    Musician(String name) {
+        this.name = name;
+    }
+
+    abstract void playInstrument();
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Musician musician = (Musician) o;
+        return Objects.equals(name, musician.name);
+    }
+
+    void printInfo() {
+        System.out.println("Name: " + name);
+    }
+}
+
+class Guitarist extends Musician {
```

Comment:
From the code point of view, what is the difference between classes `Guitarist`, `Pianist`, and `Violinist`? Do we really need to have separate classes for them? Just for comparison: classes that extend `MusicalInstruments` have different implementations of the method `playSound`, so it's fine to keep the hierarchy there.